<a href="https://colab.research.google.com/github/Sebastiandelgado14/programacion1/blob/main/QuizSebastianDelgado.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Hailer Sebastián Delgado Jerez - 2230128


# Quiz 26 de marzo

In [31]:
%%writefile pedirnotas.py
def pedir_nombre ():
  return input("Ingrese el nombre del estudiante: ")

def pedir_notas ():
  n1 = float(input("Nota 1: "))
  n2 = float(input("Nota 2: "))
  n3 = float(input("Nota 3: "))
  n4 = float(input("Nota 4: "))
  return n1, n2, n3, n4

Overwriting pedirnotas.py


In [25]:
import pedirnotas
pedirnotas.pedir_nombre()

Ingrese el nombre del estudiante: Sebastian 


'Sebastian '

In [36]:
import importlib
importlib.reload(pedirnotas)
notas = pedirnotas.pedir_notas()

Nota 1: 3.0
Nota 2: 3.0
Nota 3: 5.0
Nota 4: 4.5


In [41]:
%%writefile promedio.py
import statistics

def calcular_promedio(n1, n2, n3, n4):
    return (n1 + n2 + n3 + n4) / 4

def mediana(n1, n2, n3, n4):
    return statistics.median([n1, n2, n3, n4])

def moda(n1, n2, n3, n4):
    try:
        return statistics.mode([n1, n2, n3, n4])
    except:
        return "No hay moda única"

def estado(promedio):
    if promedio >= 3.0:
        return "Aprobado"
    else:
        return "No aprobado"

Overwriting promedio.py


In [42]:
import pedirnotas
import promedio


nombre = pedirnotas.pedir_nombre()
n1, n2, n3, n4 = pedirnotas.pedir_notas()

prom = promedio.calcular_promedio(n1, n2, n3, n4)
med = promedio.mediana(n1, n2, n3, n4)
mod = promedio.moda(n1, n2, n3, n4)
est = promedio.estado(prom)


print(" RESULTADOS ")
print("Estudiante:", nombre)
print("Notas:", n1, n2, n3, n4)
print("Promedio:", prom)
print("Mediana:", med)
print("Moda:", mod)
print("Estado:", est)

Ingrese el nombre del estudiante: Sebastian 
Nota 1: 3
Nota 2: 3
Nota 3: 5
Nota 4: 4.5
 RESULTADOS 
Estudiante: Sebastian 
Notas: 3.0 3.0 5.0 4.5
Promedio: 3.875
Mediana: 3.75
Moda: 3.0
Estado: Aprobado


El estudiante Sebastian presenta un promedio de 3.875, lo cual indica un rendimiento académico aceptable, puesto que supera el mínimo de nota para aprobar el curso, aunque no tiene una nota muy destacada.

La mediana es 3.75, lo que sugiere que la mayoría de las notas se encuentran en un rango medio, respecto a la nota aprobatoria.

La moda es 3.0, lo que indica que esta fue la calificación que más repitió el estudiante.

En general, aunque hay una nota alta la cúal es de 5.0, también existen notas más bajas, lo que evidencia cierta variabilidad en el desempeño del estudiante, pese a ello el estudiante logra aprobar el curso.